### PR TITLE
make sure exec matches the passed node type

### PR DESCRIPTION
**Pull request by dryajov**

### CODE REVIEW

**Comment by dryajov** [comment]
@diasdavid can we get this one merged, its needed for https://github.com/ipfs/interop/pull/5

File: package.json [hunk 1]
```diff
@@ -93,6 +92,7 @@
     "dirty-chai": "^2.0.1",
     "go-ipfs-dep": "0.4.13",
     "ipfs": "^0.27.5",
+    "ipfs-api": "^17.3.0",
```
Comment (diasdavid) [inline package.json]: @dryajov only realized after that ipfs-api is now a devDep. Does this mean that folks have to pass ipfs-api everytime they want to spawn a node?

Comment (diasdavid) [reply package.json]: It continues to be a prod dep https://github.com/ipfs/js-ipfsd-ctl/blob/master/src/daemon-node.js#L5

Comment (diasdavid) [reply package.json]: What was the rationale behind moving it?

**Pull request merged**